src/vivarium_gates_mncnh/components/observers.py

```diff
@@ -203,3 +231,72 @@ def load_ylds_per_case(self, builder: Builder, cause: str) -> pd.DataFrame:
         ylds = (yld_rate / incidence_rate).fillna(0).reset_index()
 
         return ylds
+
+
+class NeonatalBurdenObserver(BurdenObserver):
+    """Observer to capture death counts and ylls for neonatal sub causes."""
+
+    def __init__(self):
+        super().__init__(
+            burden_disorders=[
+                f"{NEONATAL_CAUSES.PRETERM_BIRTH}_with_rds",
+                f"{NEONATAL_CAUSES.PRETERM_BIRTH}_without_rds",
+                NEONATAL_CAUSES.NEONATAL_SEPSIS,
+                NEONATAL_CAUSES.NEONATAL_ENCEPHALOPATHY,
+                "other_causes",
+            ],
+            alive_column=COLUMNS.CHILD_ALIVE,
+            ylls_column=COLUMNS.CHILD_YEARS_OF_LIFE_LOST,
+            cause_of_death_column=COLUMNS.CHILD_CAUSE_OF_DEATH,
+        )
+
+    def register_observations(self, builder: Builder) -> None:
+        super().register_observations(builder)
+        for cause in self.burden_disorders:
+            builder.results.register_adding_observation(
+                name=f"{cause}_counts",
+                pop_filter=f"{self.cause_of_death_column} == '{cause}'",
+                requires_columns=[self.cause_of_death_column],
+                additional_stratifications=self.configuration.include,
+                excluded_stratifications=self.configuration.exclude,
+                to_observe=self.to_observe,
+            )
+
+    def to_observe(self, event: Event) -> bool:
+        return (self._sim_step_name() == SIMULATION_EVENT_NAMES.EARLY_NEONATAL_MORTALITY) or (
+            self._sim_step_name() == SIMULATION_EVENT_NAMES.LATE_NEONATAL_MORTALITY
+        )
+
+
+class NeonatalCauseRelativeRiskObserver(Observer):
```

Comment:
I'm not entirely sure if this is what we want for this observer (or even need it) but I have included it based on the docs wanting to verify stuff based on rrs.